pkg/server/handler.go

```diff
@@ -154,6 +155,13 @@ func (e *MockHandlerEngine) mockProcessing(ctx context.Context, filePath string,
 
 const MockProcessingSize = 100
 
+var MockTestFileSize = func() int64 {
+	if runtime.GOARCH == "s390x" {
+		return 4096
+	}
```

Comment:
Quick question: Can not the file be truncated to size 100B in s390x arch? 

Besides being used as a counter, this value is just the truncation size of a test file:
https://github.com/longhorn/backing-image-manager/blob/6bcb81541be8c52c0bb543cea357230d37fc87bd/pkg/server/handler.go#L143
https://github.com/longhorn/backing-image-manager/blob/6bcb81541be8c52c0bb543cea357230d37fc87bd/pkg/server/handler.go#L163

Reply:
@raulcabello was there any specific reason for the change?

Reply:
It seems there are some issues on s390x related to block sizes so the integration tests don't work with file sized less than 4096B on s390x.

See here for reference: https://suse.slack.com/archives/C02DJMVTT98/p1639738122326000?thread_ts=1638371786.214600&cid=C02DJMVTT98

Reply:
Then you can directly modify the constant to 4096 instead, then the arch check is unnecessary.

Reply:
I tried to do as you requested but unfortunately it breaks a test: https://drone-pr.longhorn.io/longhorn/backing-image-manager/89/1/3.

I tested manually and setting MockProcessingSize to 4096 but leaving the loop counter at 100 works.

Would that change be Ok for you?

Reply:
@thomasferrandiz we were having problems with the file size in the tests for this kernel issue on s390x https://github.com/longhorn/longhorn/issues/3733 , that's why it was working with 4096 and not with 100. It should work with 100 once you use the latest [kernel-default-5.14.21-150400.12.2](https://github.com/longhorn/longhorn/issues/3733#issuecomment-1077597692)

Reply:
Given @raulcabello's answer above, I reverted the commit so that we still use the arch check in the tests.
This seems the best way while we wait for the fixed kernel to become the default.

Reply:
I just help update the test file size in https://github.com/longhorn/backing-image-manager/pull/50. Then you don't need to worry about this tedious issue.

Reply:
> I just help update the test file size in #50. Then you don't need to worry about this tedious issue.

Thanks @shuo-wu .
I will wait for your PR to be merged then rebase on it.

Reply:
@thomasferrandiz @shuo-wu PR merged, please rebase again. Thanks.

Reply:
I rebased on PR 50 but unfortunately it didn't help because PR50 changes `MockProcessingSize` in `pkg/datasource` and the issue was in `pkg/server.`

I tried a similar fix however and all the tests are Ok.
I hope that the change is fine.

Reply:
Sorry for missing that part. The change you add looks good!